MapboxCoreNavigation/RouteController.swift

```diff
@@ -675,6 +670,27 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    /**
+     Monitors the user's course to see if it is consistantly moving away from what we expect the course to be at a given point.
+     */
+    func userCourseIsOnRoute(_ location: CLLocation) -> Bool {
+        let nearByCoordinates = routeProgress.currentLegProgress.nearbyCoordinates
+        
+        guard let calculatedCourseForLocationOnStep = interpolatedCourse(from: location, along: nearByCoordinates) else { return true }
```

Comment:
This is similar enough to the existing dead reckoning code that we should consider how they interact:

https://github.com/mapbox/mapbox-navigation-ios/blob/08bd523fd9c9fc0e045e1635baa88e573dc21dbc/MapboxCoreNavigation/RouteController.swift#L686-L688
https://github.com/mapbox/mapbox-navigation-ios/blob/b381563387262b408dfe1939883c813bd616654b/MapboxCoreNavigation/Constants.swift#L40

The dead reckoning code reroutes the user based on where they’ll be in one second, which is typically when the next location update arrives, based on the current speed.